lib/fitfree.gi

```diff
@@ -28,7 +28,17 @@ local cache,ffs,pcisom,rest,it,kpc,k,x,ker,r;
 
   pcisom:=ffs.pcisom;
 
-  rest:=RestrictedMapping(ffs.factorhom,U);
+  #rest:=RestrictedMapping(ffs.factorhom,U);
+  if IsPermGroup(U) and AssertionLevel()>1 then
+    rest:=GroupHomomorphismByImages(U,Range(ffs.factorhom),GeneratorsOfGroup(U),
+      List(GeneratorsOfGroup(U),x->ImagesRepresentative(ffs.factorhom,x)));
+  else
+    RUN_IN_GGMBI:=true; # hack to skip Nice treatment
+    rest:=GroupHomomorphismByImagesNC(U,Range(ffs.factorhom),GeneratorsOfGroup(U),
+      List(GeneratorsOfGroup(U),x->ImagesRepresentative(ffs.factorhom,x)));
+    RUN_IN_GGMBI:=false;
+  fi;
+  if rest=fail then Error("can't build homomorphism"); fi;
```

Comment:
So when would this ever fail? Can we perhaps give a slightly more helpful error message?

Reply:
I've replaced the error with an assertion.